ttnn/cpp/ttnn/operations/moreh/moreh_matmul/device/moreh_matmul_device_operation.hpp

```diff
@@ -0,0 +1,84 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <optional>
+
+#include "ttnn/decorators.hpp"
+#include "ttnn/device_operation.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/types.hpp"
+
+namespace ttnn::operations::moreh::moreh_matmul {
+struct MorehMatmulOperation {
+    struct operation_attributes_t {
+        bool transpose_input;
+        bool transpose_other;
+
+        const std::optional<MemoryConfig> output_memory_config;
```

Comment:
```suggestion
        const MemoryConfig output_memory_config;
```
Please do not use optional for `MemoryConfig` within `operation_attributes_t`.